OOInteraction/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -0,0 +1,155 @@
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#include "CAddNodeToViewByName.h"
+
+#include "OOModel/src/declarations/Declaration.h"
+#include "VisualizationBase/src/items/ViewItem.h"
+#include "OOModel/src/declarations/Declaration.h"
+#include "OOModel/src/declarations/Class.h"
+#include "OOModel/src/declarations/Method.h"
+#include "OOModel/src/declarations/Module.h"
+#include "OOModel/src/declarations/Project.h"
+
+#include "ModelBase/src/model/TreeManager.h"
+
+
+namespace OOInteraction {
+
+CAddNodeToViewByName::CAddNodeToViewByName() : Command{"add"}{}
+
+bool CAddNodeToViewByName::canInterpret(Visualization::Item*, Visualization::Item*,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>&)
+{
+	return commandTokens.size() >= 1 && commandTokens.first() == name();
+}
+
+Interaction::CommandResult* CAddNodeToViewByName::execute(Visualization::Item* source, Visualization::Item*,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>&)
+{
+	if (commandTokens.size() < 2)
+		return new Interaction::CommandResult(new Interaction::CommandError("Please specify a node to add"));
+
+	if (auto node = findNode(commandTokens[1],
+			Model::AllTreeManagers::instance().loadedManagers().first()->root()))
+	{
+		source->scene()->currentViewItem()->insertNode(node);
+		return new Interaction::CommandResult();
+	}
+	else
+		return new Interaction::CommandResult(new Interaction::CommandError(
+					"Could not find node with name " + commandTokens[1]));
+}
+
+QList<Interaction::CommandSuggestion*> CAddNodeToViewByName::suggest(Visualization::Item*, Visualization::Item*,
+		const QString& textSoFar, const std::unique_ptr<Visualization::Cursor>&)
+{
+	QList<Interaction::CommandSuggestion*> suggestions;
+
+	if (textSoFar.trimmed().startsWith("add ", Qt::CaseInsensitive))
+	{
+		auto nodeName = textSoFar.trimmed().mid(4);
+		auto names = findNames(nodeName.split("_"),
```

Comment:
we should use . instead of _. I think the reason why it might not have worked is that . (unlike _ ) results in a separate token: so `a_b` is one token whereas `a.b` are three tokens. Could you try this and see if you can adapt your code?